libraries/standard/mqtt/integration-test/mqtt_system_test.c

```diff
@@ -832,19 +855,88 @@ void test_MQTT_Connect_LWT( void )
     TEST_ASSERT_TRUE( receivedUnsubAck );
 }
 
+/**
+ * @brief Verifies that the MQTT library sends a Ping Request packet if the connection is
+ * idle for more than the keep-alive period.
+ */
 void test_MQTT_ProcessLoop_KeepAlive( void )
 {
     uint32_t connectPacketTime = context.lastPacketTime;
     uint32_t elapsedTime = 0;
+
     TEST_ASSERT_EQUAL( 0, context.pingReqSendTimeMs );
 
-    /* Sleep until control packet needs to be sent. */
+    /*Sleep until control packet needs to be sent. */
     Clock_SleepMs( MQTT_KEEP_ALIVE_INTERVAL_SECONDS * 1000 );
     TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_ProcessLoop( &context, MQTT_PROCESS_LOOP_TIMEOUT_MS ) );
 
     TEST_ASSERT_NOT_EQUAL( 0, context.pingReqSendTimeMs );
     TEST_ASSERT_NOT_EQUAL( connectPacketTime, context.lastPacketTime );
-    /* Test that the ping was sent within 1.5 times the keep alive interval. */
+    /*Test that the ping was sent within 1.5 times the keep alive interval. */
     elapsedTime = context.lastPacketTime - connectPacketTime;
     TEST_ASSERT_LESS_OR_EQUAL( MQTT_KEEP_ALIVE_INTERVAL_SECONDS * 1500, elapsedTime );
 }
+
+/**
+ * @brief Verifies the behavior of the MQTT library in a restored session connection with the broker.
+ * Tests that the library resends PUBREL packets to the broker in a restored session for an incomplete
+ * PUBLISH operation in a previous connection.
+ */
+void test_MQTT_Connect_Restore_Session( void )
+{
+    /* Graceless disconnect. Terminate TLS session and TCP connection. */
+    ( void ) Openssl_Disconnect( &networkContext );
```

Comment:
I don't think a graceless disconnect is required, there's just little point in sending the disconnect packet if the session will be discarded anyway. Can you edit the comment to clarify that?

Reply:
Good point. Updated